### ISSUE TITLE
Feat: Adding Oauth2 register and login using appwrite

### ISSUE THREAD

**Issue by TejasNasre** [Sanchitbajaj02/palettegram#284]
### What feature?

# Oauth2 Register and login in app using appwrite by which user experience will increase and i faced some issues with the register like expiring session so by google login user can easily login and start to use app.  

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I want to work on this issue

**Comment by Sanchitbajaj02** [comment]
> ### What feature?
> Oauth2 Register and login in app using appwrite by which user experience will increase and i faced some issues with the register like expiring session so by google login user can easily login and start to use app.

Hi @TejasNasre 
Google and GitHub has already been in configured but not up and running due to some server challenges and connection problems.
If you're interested you can checkout the scope checking and how to make that process bug free.

> [!NOTE] Will assign that issue tomorrow after merging the Github OAuth PR

**Comment by TejasNasre** [comment]
Yeah I saw that you added google Oauth2 in applications but not working because of the provider issue or something so I can fix this bug after your PR I will test and if any problem then you can assign this issue to me.

**Comment by TejasNasre** [comment]
Thanks for assigning issue @Sanchitbajaj02 i will start working on issue soon

**Comment by TejasNasre** [comment]
Hey @Sanchitbajaj02 i tried solving issue but it not work and i am getting difficulty in working with next and typescript.I recently implemented the feature in my personal project but i have not worked with next and typescript before so the feature is getting errors.

you can assign issue to someone who can work on this or delete this issue.Thanks for understanding.

**Comment by thecloudcode** [comment]
Hi @Sanchitbajaj02, I have prior work experience in Next.js. Kindly, assign me this issue, I would love to work on it.

**Comment by Sanchitbajaj02** [comment]
> Hi @Sanchitbajaj02, I have prior work experience in Next.js. Kindly, assign me this issue, I would love to work on it.

Sorry for the late reply. Assigned to you

**Comment by thecloudcode** [comment]
@Sanchitbajaj02 thank you for assigning

**Comment by Sanchitbajaj02** [comment]
Since, I didn't hear from a contributor, I am unassigning it
